executor/include/execution/trait/can_require.hpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by Shrijit Singh on 14/06/20.
+//
+
+#pragma once
+
+#include <type_traits>
+
+namespace execution {
+
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor> &&
+                  Property::template static_query<Executor>(),
+              int> = 0>
+Executor require(Executor ex, const Property t) {
+  return ex.require(t);
+}
+
+template <typename Executor, typename Properties, typename = void>
+struct can_require : std::false_type {};
```

Comment:
What's the difference between `is_applicable_property` and `can_require`?

Reply:
`is_applicable_property` is used to restrict the mentioned properties to executors to only executors and not any type.

`can_require` checks whether the specified executor supports the specified require customization point.